src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
 export interface Figure {
-
+  shape: string,
+  color: string,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape;
 
-}
+  public sides;
+
+  constructor(public color: 'red' | 'green' | 'blue', ...args: number[]) {
+    this.shape = 'triangle';
+    this.color = color;
+    this.sides = [...args].sort(); /* Sort helps with detecting 0s if they
+                                      were not in the first position */
 
-export class Circle {
+    if ((this.sides[0] + this.sides[1] <= this.sides[2])
+    || (this.sides[2] + this.sides[0] <= this.sides[1])
+    || (this.sides[2] + this.sides[1] <= this.sides[0])
+    || this.sides[0] <= 0) {
+      throw new Error('Triangle not possible with given sides!');
+    }
+  }
 
+  getArea(): number {
+    /* I hate math and therefore hate you for this,
+    but as requested, Heron's formula */
+    const semiPer: number = (this.sides[0] + this.sides[1] + this.sides[2]) / 2;
+    const bracket1: number = semiPer - this.sides[0];
+    const bracket2: number = semiPer - this.sides[1];
+    const bracket3: number = semiPer - this.sides[2];
+    const squaredResult: number = semiPer * bracket1 * bracket2 * bracket3;
+    const result: number = Math.sqrt(squaredResult);
+
+    return (Math.round(result * 100)) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape;
+
+  constructor(public color: 'red' | 'green' | 'blue', public radius: number) {
+    this.shape = 'circle';
+    this.color = color;
+    this.radius = radius;
```

Comment:
checklist 5

Reply:
But... the checklist only has 3 points in it

Reply:
1. [CODE STYLE] - ```getArea``` function doesn't need to accept any parameters. All necessary values are already available as object fields
2. [CODE STYLE] - Enum constant names usually start with capital letter. Values can have any case that's necessary.

BAD EXAMPLE:
```
export enum Directions {
  up = 'UP',
  down = 'DOWN',
  left = 'LEFT',
  right = 'RIGHT',
}
```

GOOD EXAMPLE:
```
export enum Directions {
  Up = 'UP',
  Down = 'DOWN',
  Left = 'LEFT',
  Right = 'RIGHT',
}
```

3. [CODE STYLE] - If you throw an error try to provide its detailed description in error message.

BAD EXAMPLE:
```
if (x % 2 !== 0) {
  throw new Error('Something went wrong')
}
```

GOOD EXAMPLE:
```
if (x % 2 !== 0) {
  throw new Error('Resulting value should be even number')
}
```



That's.. the whole file

Reply:
you are right, I added this to the wrong file...
it's point 4 right now